File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] result = new int[nums.Length];
 int product = 1 , product1 = 1 , cnt0 = 0;
 for (int i = 0; i < nums.Length; i++)
 {
     if (nums[i] != 0)
     {
         product *= nums[i];
     }
     else
         cnt0++;
     product1 *= nums[i];
 }
 for(int i = 0; i < nums.Length; i++)
 {
     if(cnt0 > 1)
     {
         result[i] = 0;
     }else if(product != product1 && nums[i] == 0)
     {
         result[i] = product;
     }else
     {
         result[i] = product1 / nums[i];
     }
 }
 return result;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
         int cnt0 = 0, product = 1, j = 0;
        int[] res = new int[nums.length];
        for(int i = 0; i <nums.length ; i++ ){
            if(nums[i] == 0){
                cnt0++;
                j = i;
            }else
                product *= nums[i];
        }
        if(cnt0 > 1){
            return res;
        }else if (cnt0 == 1){
            res[j] = product;
        }else{
            for(int i = 0;i<nums.length;i++){
                res[i] = product / nums[i];
            }
        }
        return res;
    }
}